packages/commerce/backend/src/order/controller.ts

```diff
@@ -80,4 +81,39 @@ export class OrderController implements IOrderController {
       next(err)
     }
   }
+
+  public async finish(req: Request, res: TypedResponse, next: NextFunction) {
+    try {
+      const orderId = req.params.id
+      if (!orderId) {
+        throw new BadRequest('Order ID was not provided.')
+      }
+
+      // TODO: Verify hash
+      // https://rafiki.dev/concepts/open-payments/grant-interaction/#endpoints
+      const { interactRef, result } = await validate(
+        finishOrderSchema,
+        req.body
+      )
+
+      const order = await this.orderService.ensurePendingState(orderId)
+
+      if (result) {
+        const status = result === 'grant_rejected' ? 200 : 400
```

Comment:
introducing a separate const for rejected state may make this a bit more readable
```
        const isRejected = result === 'grant_rejected'
        const status = isRejected ? 200 : 400
        const message = isRejected ? 'SUCCESS' : 'FAILED'
        this.orderService.reject(orderId)
        res
          .status(status)
          .json({ success: isRejected, message })